Source/WebCore/page/SecurityOriginData.h

```diff
@@ -36,49 +36,88 @@ namespace WebCore {
 class Frame;
 class SecurityOrigin;
 
+enum OpaqueOriginIdentifierType { };
+using OpaqueOriginIdentifier = ObjectIdentifier<OpaqueOriginIdentifierType>;
+
 struct SecurityOriginData {
-    enum OpaqueOriginIdentifierType { };
-    using OpaqueOriginIdentifier = ProcessQualified<ObjectIdentifier<OpaqueOriginIdentifierType>>;
-    using MarkableOpaqueOriginIdentifier = Markable<OpaqueOriginIdentifier, OpaqueOriginIdentifier::MarkableTraits>;
+    struct Tuple {
+        String protocol;
+        String host;
+        std::optional<uint16_t> port;
+        bool operator==(const Tuple& other) const { return protocol == other.protocol && host == other.host && port == other.port; }
+        Tuple isolatedCopy() const { return { protocol.isolatedCopy(), host.isolatedCopy(), port }; }
```

Comment:
We should have an isolatedCopy() version that works on a r-value reference so that it can be leverage by the one we have on SecurityOriginData.